tests/e2e/provisioning/pkg/client/kyma/kyma_client.go

```diff
@@ -0,0 +1,69 @@
+package kyma
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Client contains logic allowing to check if Kyma instance is accessible
+type Client struct {
+	client http.Client
+	log    logrus.FieldLogger
+}
+
+func NewClient(clientHttp http.Client, log logrus.FieldLogger) *Client {
+	clientHttp.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+	return &Client{
+		client: clientHttp,
+		log:    log,
+	}
+}
+
+// CallDashboardURL sends request to the dashboardURL and expects to be redirected to the logging page
+// There are 2 possibility locations where we can be redirected:
+// `/auth/xsuaa` - if uua-issuer exist on the cluster
+// `/auth/local` - if Kyma use a standard login page
+func (c *Client) CallDashboardURL(dashboardURL string) error {
+	targetURL := c.buildTargetURL(dashboardURL)
+
+	c.log.Infof("Calling the dashboard URL: %s", targetURL)
+	resp, err := c.client.Get(targetURL)
+	if err != nil {
+		return errors.Wrapf(err, "while calling dashboard '%s'", dashboardURL)
+	}
+	defer c.warnOnError(resp.Body.Close())
+
+	if resp.StatusCode != http.StatusFound {
+		return errors.Errorf("expected status code: %d, got %d", http.StatusFound, resp.StatusCode)
+	}
+	if location, err := resp.Location(); err != nil {
+		return errors.Wrap(err, "while getting response location")
+	} else if location.Path != "/auth/xsuaa" {
+		return errors.Errorf("request was wrongly redirected: %s", location.String())
+	}
+	c.log.Info("Successful response from the dashboard URL")
+
+	return nil
+}
+
+// Kyma console URL won't redirect us to the UUA logging page, to achieve that we must call dex with a set of parameters
+// state and nonce params are faked
+func (c *Client) buildTargetURL(dashboardURL string) string {
+	domain := strings.Split(strings.Split(dashboardURL, "console.")[1], "/")[0]
+	params := "&response_type=id_token%20token&scope=audience%3Aserver%3Aclient_id%3Akyma-client%20audience%3" +
+		"Aserver%3Aclient_id%3Aconsole%20openid%20profile%20email%20groups&state=5a4f3d15&nonce=79bf9c29"
+
+	return fmt.Sprintf("https://dex.%s/auth?client_id=console&redirect_uri=%s%s", domain, dashboardURL, params)
```

Comment:
maybe something more readable :) 
```
u := url.URL{
  Scheme: "https",
  Host: fmt.Sprintf("dex.%s/auth", domain),
}
q := u.Query()
q.Set("client_id", "console")
q.Set("redirect_uri", dashboardURL)
q.Set("response_type", "id_token")
q.Set("scope", "audience")
(...)
u.RawQuery = q.Encode()

return u.String()
```

Reply:
same comment

Reply:
I discussed this with goats and they believe we don't have a better option to assert if UUA works fine. There was one idea to use some http client with js interpreter to just call dashboardURL and behave like a browser but i didn't find anything interesting.

I changed this code accordingly to your proposition.